hal/halx86/legacy/bus/pcibus.c

```diff
@@ -311,6 +311,36 @@ HalpWritePCIConfig(IN PBUS_HANDLER BusHandler,
     }
 }
 
+#ifdef SARCH_XBOX
+BOOLEAN
+NTAPI
+HalpXboxBlacklistedPCISlot(IN PBUS_HANDLER BusHandler,
+                           IN PCI_SLOT_NUMBER Slot)
+{
+    /* Trying to get PCI config data from devices 0:0:1 and 0:0:2 will completely
+     * hang the Xbox. Also, the device number doesn't seem to be decoded for the
+     * video card, so it appears to be present on 1:0:0 - 1:31:0.
+     * We hack around these problems by indicating "device not present" for devices
+     * 0:0:1, 0:0:2, 1:1:0, 1:2:0, 1:3:0, ...., 1:31:0 */
+    if ((BusHandler->BusNumber == 0 && Slot.u.bits.DeviceNumber == 0 &&
```

Comment:
Might wanna create a simple table for this if there are more devices / ranges (in a next pr)